File: design-patterns/java/java-adapter-design-pattern/MssqlAdapter.java
// The MssqlAdapter class adapts the Mssql class to the DB interface.
public class MssqlAdapter implements DB{
    private Mssql mssql;

    public MssqlAdapter(){
        mssql = new Mssql();
    }

    @Override
    public User selectUserById(int id) {
       return mssql.userById(id);
    }

    @Override
    public void insertUser(User user) {
        mssql.user(user);
    }
   

}

File: design-patterns/python/python-adapter-design-pattern/MssqlAdapter.py
from DB import DB
from Mssql import Mssql

# The MssqlAdapter class adapts the Mssql class to the DB interface.
class MssqlAdapter(DB):
    def __init__(self):
        self.mssql = Mssql()

    def select_user_by_id(self, id):
        return self.mssql.user_by_id(id)

    def insert_user(self, user):
        self.mssql.user(user)    
